model/task/task.go

```diff
@@ -561,6 +562,92 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 
 }
 
+func (t *Task) UpdateDisplayTask() error {
+	if !t.DisplayOnly {
+		return fmt.Errorf("%s is not a display task", t.Id)
+	}
+
+	statuses := []string{}
+	catcher := grip.NewBasicCatcher()
+	var timeTaken time.Duration
+	for _, taskId := range t.ExecutionTasks {
+		execTask, err := FindOne(ById(taskId))
+		catcher.Add(err)
+
+		// if any of the execution tasks are scheduled, the display task is too
+		if execTask.Activated {
+			t.Activated = true
+		}
+
+		// the display task's status will be the highest priority of its exec tasks
+		if execTask != nil {
+			statuses = append(statuses, execTask.ResultStatus())
+		}
+
+		// add up the duration of the execution tasks
+		timeTaken += execTask.TimeTaken
+	}
+	if catcher.HasErrors() {
+		return catcher.Resolve()
+	}
+	if len(statuses) > 0 {
+		sort.Sort(byPriority(statuses))
+		t.Status = statuses[0]
+	}
+
+	t.TimeTaken = timeTaken
+
+	return UpdateOne(
```

Comment:
mgo has an UpdateId method, do our DB wrappers also expose this? I don't think it matters much but it might be more clear in circumstances like this.

Reply:
We do have one, though not in the task DB model. I don't think it makes enough of a difference to add one